pkg/v1/sdk/features/plugin/runner.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	k8sconfig "sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	configv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/config/v1alpha1"
+)
+
+type featureGateRunner struct {
+	featureGateClient client.Client
+}
+
+// newFeatureGateRunner returns an instance of featureGateRunner.
+func newFeatureGateRunner() (*featureGateRunner, error) {
+	featureGateClient, err := getFeatureGateClient()
+	if err != nil {
+		return nil, err
+	}
+	return &featureGateRunner{featureGateClient: featureGateClient}, nil
+}
+
+// getFeatureGateClient returns a new FeatureGate client
+func getFeatureGateClient() (client.Client, error) {
+	var restConfig *rest.Config
+	var err error
+
+	scheme := runtime.NewScheme()
+	if err := configv1alpha1.AddToScheme(scheme); err != nil {
+		return nil, err
+	}
+	if err := corev1.AddToScheme(scheme); err != nil {
+		return nil, err
+	}
+
+	if kubeConfig == "" {
+		if restConfig, err = k8sconfig.GetConfig(); err != nil {
+			return nil, err
+		}
+	} else {
+		config, err := clientcmd.LoadFromFile(kubeConfig)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load from %s: %w", kubeConfig, err)
+		}
+		rawConfig, err := clientcmd.Write(*config)
+		if err != nil {
+			return nil, fmt.Errorf("unable to write config: %w", err)
+		}
+		if restConfig, err = clientcmd.RESTConfigFromKubeConfig(rawConfig); err != nil {
+			return nil, fmt.Errorf("unable to set up rest config: %w", err)
+		}
+	}
+	mapper, err := apiutil.NewDynamicRESTMapper(restConfig, apiutil.WithLazyDiscovery)
```

Comment:
IIRC, controller-runtime client already defaults to DynamicRESTMapper, so this is probably not necessary.